bank: deposit & withdraw transactions

BorislavAnchev opened the pull request:
The user is now able to perform deposit and withdraw transactions
to his bank accounts.

BorislavAnchev commented:
@mgenov, new code ready for a review.

BorislavAnchev commented:
Changes available for a review:
- Added test for the submit warning paragraph when all the required fields are filled out and when warning shouldn't be shown
- Added test for the 'Current balance' paragraph

BorislavAnchev commented:
@mgenov, could you take a look when available?

mgenov commented:
ok, will take a look into it tomorrow morning.

BorislavAnchev commented:
@mgenov, just reminding for the review.

mgenov commented on banktask/src/redux/middlewares/axios.js:
```diff
@@ -0,0 +1,21 @@
+        type: `${action.type}_SUCCESS`,
+        payload: response.data
+      });
+      if(action.meta && action.meta.alert) {
```
doesn't look reasonable UI element as an alert to be used in a middleware.

BorislavAnchev replied:
Where is the correct place to put it? Since I want the alert to be shown only after successful response, I could put it into the reducer, since it will reach the _SUCCESS case only when the account was successfully deleted, but it seems out of place to me there.

mgenov commented on banktask/src/components/Button/Button.js:
```diff
@@ -0,0 +1,22 @@
+import PropTypes from 'prop-types';
+import './styles.scss';
+
+const Button = ({buttonText, onClick, 'data-testid': dataTestId}) => {
```
buttonText could be named children as it will let the caller is it like:

```jsx
<Button>Submit</Button>
```

mgenov commented on banktask/src/components/Button/Button.spec.js:
```diff
@@ -0,0 +1,51 @@
+    'data-testid': 'Test Button'
+  };
+
+  afterEach(cleanup);
```
it is better afterEach to be defined first for clearance.

BorislavAnchev commented:
New changes ready for a review:

- swapped the 'buttonText' component property for the 'children' property in the <Button> component
- Moved the 'afterEach' method at the beginning of the tests and added it to some tests that didn't have one
- The alert is removed from the middleware and is not being shown for now

BorislavAnchev commented:
@mgenov, code ready for a review.
